### DISCUSSION TITLE
How can I contribute to the Tcases project on proposed ideas?

### DISCUSSION THREAD

**Discussion by arunveersingh** [Cornutum/tcases#283] [Ideas]
Dear Project Maintainers,

I'm Arunveer Singh, a developer and software architect specializing in Java Enterprise Applications. I've been following your project and am truly impressed with the capabilities of the project. It's quite relevant in real world scenarios.

I've identified a potential enhancement to further optimize its usability for wider production ready scenarios:

Idea 1:
**Current State**: When we have 3 open API specifications, Tcases generates three distinct JUnit files, each using the same API key from the environment.
**Challenge**: If these 3 APIs (OAS) are deployed on API gateways, they might necessitate three individual API keys.
**Proposed Enhancement**: Implement a mechanism that allows for different API keys from the environment, defaulting to the main API key when one isn't available. This adjustment will maintain compatibility with existing setups.

Idea 2:
**Current State**: Project provides maven plugin.
**Challenge**: Projects using gradle find it little challenging to integrate within existing pipelines.
**Proposed Enhancement**: Building a gradle plugin.

Having used Tcases in few POCs and appreciating its OAS integration (able to catch few bugs in our OAS and it's implementation), I'm eager to contribute. Could you guide me on how to proceed, whether by direct contribution or through a fork?

Thank you for your consideration. I look forward to your feedback.

Best regards,
Arunveer Singh

**Comment by kerrykimbrough** [comment]
Hi, Arunveer

Your contributions to Tcases will be most welcome. Here's what I suggest.

First, submit an issue for each enhancement you'd like to see. Be sure to explain exactly what is needed and why it is needed, much like you have already done. Then, in your own fork of the tcases repo, create a new branch for each issue you intend to work on. When your changes are complete in the branch, create a pull request to merge your changes into the master repo.

About Idea 1: I'm not sure I understand the problem you intend to solve. I'm not sure what you mean by an "API key in the environment". When you submit an issue for this, can you include more explanation?
